### PR TITLE
unrar: strip out native compilation flag

### PR DESCRIPTION
**Pull request by Zapeth**
#### Testing the changes
- I tested the changes in this PR: **briefly**

This should get rid of the native compilation flag in the makefile script.

Not sure if it was included upstream by mistake, but it should be looked out for when new versions are released (ie revert this change if the flag gets removed again).

### CODE REVIEW

File: srcpkgs/unrar/template [hunk 1]
```diff
@@ -12,13 +12,12 @@ checksum=505c13f9e4c54c01546f2e29b2fcc2d7fabc856a060b81e5cdfe6012a9198326
...
 do_build() {
-	vsed -e 's/^\(CXXFLAGS\)=\(.*\)/\1+=\2/' \
+	vsed -e 's/^\(CXXFLAGS\)=-march=native \(.*\)/\1+=\2/' \
 		 -e 's/^\(LDFLAGS\)=\(.*\)/\1+=-lpthread \2/' \
 		 -i makefile
 	make CXX="$CXX" LD="$LD" STRIP=: -f makefile
-	# early install of unrar because makefile deletes unrar on lib creation
+	# early install of unrar because need to clean for lib creation
 	vbin unrar
-	# recompile for lib
```
Comment (vincele) [inline srcpkgs/unrar/template:21]: same

File: srcpkgs/unrar/template [hunk 2]
```diff
@@ -12,13 +12,12 @@ checksum=505c13f9e4c54c01546f2e29b2fcc2d7fabc856a060b81e5cdfe6012a9198326
 repository=nonfree
 
 do_build() {
-	vsed -e 's/^\(CXXFLAGS\)=\(.*\)/\1+=\2/' \
+	vsed -e 's/^\(CXXFLAGS\)=-march=native \(.*\)/\1+=\2/' \
 		 -e 's/^\(LDFLAGS\)=\(.*\)/\1+=-lpthread \2/' \
 		 -i makefile
 	make CXX="$CXX" LD="$LD" STRIP=: -f makefile
-	# early install of unrar because makefile deletes unrar on lib creation
+	# early install of unrar because need to clean for lib creation
```
Comment (vincele) [inline srcpkgs/unrar/template:19]: This does not look related to patch subject

File: srcpkgs/unrar/template [hunk 3]
```diff
@@ -12,13 +12,12 @@ checksum=505c13f9e4c54c01546f2e29b2fcc2d7fabc856a060b81e5cdfe6012a9198326
 repository=nonfree
 
 do_build() {
-	vsed -e 's/^\(CXXFLAGS\)=\(.*\)/\1+=\2/' \
+	vsed -e 's/^\(CXXFLAGS\)=-march=native \(.*\)/\1+=\2/' \
```
Comment (vincele) [inline srcpkgs/unrar/template:15]: This is assuming  '-march=native' is always the first in CXXFLAGS
Is that a safe assumption ?

Comment (Zapeth) [reply srcpkgs/unrar/template:15]: Yes, for version 7.0.9 it is (upstream makefile has it hardcoded like that), I wouldn't expect them to re-release the same version with the flag (re)moved.

File: srcpkgs/unrar/template (hunk 2 above)
Comment (Zapeth) [reply srcpkgs/unrar/template:19]: Its just a small comment correction (along with the other comment change), but if thats not appropriate to include in the PR then I'll remove it.

Comment (vincele) [reply srcpkgs/unrar/template:19]: I'm no authority on the subject, that was just a comment while passing by, feel free to ignore

File: srcpkgs/unrar/template (hunk 3 above)
Comment (vincele) [reply srcpkgs/unrar/template:15]: OK, as-is it' no problem, but if upstream changes that in the future, it may be a (slight) trap for the one updating the void packaging. But that does not look like a big problem, just if it does not match the regex, the flag will be used again... Feel free to ignore this comment as well.
